src/sentry/utils/performance_issues/performance_detection.py

```diff
@@ -907,33 +911,48 @@ def _validate_and_store_performance_problem(self):
         exceeds_count_threshold = len(self.consecutive_db_spans) >= self.settings.get(
             "consecutive_count_threshold"
         )
-        if exceeds_count_threshold and self._are_db_spans_independant(self.consecutive_db_spans):
+        independant_db_spans = self._find_independant_spans(self.consecutive_db_spans)
+        exceeds_duration_threshold = self._sum_span_duration(
+            independant_db_spans
+        ) > self.settings.get("duration_threshold")
+        if exceeds_count_threshold and exceeds_duration_threshold:
             self._store_performance_problem()
 
     def _store_performance_problem(self) -> None:
         fingerprint = self._fingerprint()
+        offender_span_ids = list(
+            map(lambda span: span.get("span_id", None), self.consecutive_db_spans)
+        )
         self.stored_problems[fingerprint] = PerformanceProblem(
             fingerprint,
             "db",
             "consecutive db",
             GroupType.PERFORMANCE_CONSECUTIVE_DB_OP,
             cause_span_ids=None,
             parent_span_ids=None,
-            offender_span_ids=self.consecutive_db_spans,
+            offender_span_ids=offender_span_ids,
         )
 
-    def _are_db_spans_independant(self, spans: list[Span]) -> bool:
+    def _sum_span_duration(self, spans: list[Span]) -> int:
+        "Given a list of spans, find the sum of the span durations in milliseconds"
+        sum = 0
+        for span in spans:
+            sum += get_span_duration(span).total_seconds() * 1000
+        return sum
+
+    def _find_independant_spans(self, spans: list[Span]) -> list[Span]:
         """
         Given a list of spans, checks if there is at least a single span that is independent of the rest.
         To start, we are just checking for a span in a list of consecutive span without a WHERE clause
         """
+        independant_spans = []
         for span in spans[1:]:
             query: str = span.get("description", None)
-            if not query:
-                return False
-            if "WHERE" not in query.upper():  # TODO - use better regex
-                return True
-        return False
+            if (
+                query and contains_complete_query(span) and "WHERE" not in query.upper()
```

Comment:
Why do we care about complete queries for consecutive db spans? I can understand for N+1 where we need to ensure the query is the same.

Reply:
With the current version, we need to ensure that the spans are independant of one another, requiring the full query.